### PR TITLE
feat: add reusable http client

### PR DESCRIPTION
**Pull request by jarrvis**

### CODE REVIEW

**Review by kgracki — APPROVED** [review]
🚀

File: src/api/http.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import Agent from 'agentkeepalive';
+const HttpsAgent = Agent.HttpsAgent;
+import axios from "axios";
+
+const keepAliveAgent = new Agent({
+    maxSockets: 160,
```
Comment (kgracki) [inline src/api/http.ts:6]: awesome! just to know - are these values used from some good practices or do we need to treat them experimentally?

Comment (jarrvis) [reply src/api/http.ts:6]: hey, here is the resource used:
https://azureossd.github.io/2022/03/10/NodeJS-with-Keep-Alives-and-Connection-Reuse/index.html

and here it is referenced in axios ETIMEDOUT error context:
https://stackoverflow.com/questions/63064393/getting-axios-error-connect-etimedout-when-making-high-volume-of-calls

The client config was tested on browser (examples/react-starter) and nodejs (jest tests).
Unfortunately I was not able to reproduce the ETIMEDOUT error before doing the changes so it's a blind try.
Ideally we need a load test with default axios config - hopefully getting some networking errors and then rerunning with this change included

Comment (jarrvis) [reply src/api/http.ts:6]: Here is a nice resource about networking issues:
https://medium.com/ssense-tech/reduce-networking-errors-in-nodejs-23b4eb9f2d83
and how they are mitigated using `agentkeepalive` package. Many can be avoided using persistent connection (if we simply get rid of TLS handshake for subsequent http calls)
List of errors:
EAI_AGAIN
ETIMEDOUT
ECONNRESET
ESOCKETTIMEOUT
ECONNREFUSED
socket hang-up.

For some of them we do have a retry already, here in SDK:
https://github.com/Akord-com/akord-js/blob/90c013fc1a16772820ed4a2edfea5e7ced844680/src/errors/error-factory.ts#L35

**Pull request merged**